### PR TITLE
Port HIVE-27815: Support update numModifiedRows

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description

Backport https://github.com/apache/hive/pull/4819

Note: it's only the JDBC driver side change, to make it work, we also need to modify the engines.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Test Plan 🧪

We can not test this feature so far because we don't have engine support that. Pass GHA to ensure it breaks nothing.

---

# Checklist 📝

- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiStatement.java [hunk 1]
```diff
@@ -590,7 +590,16 @@ public int getUpdateCount() throws SQLException {
      * finished.
      */
     waitForOperationToComplete();
```
Comment (turboFei) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiStatement.java:592]: Seems this line can be removed.

Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiStatement.java]: right, thanks for checking

**Review by turboFei — APPROVED** [review]

**Comment by pan3793** [comment]
interesting error, CI runs at mid-night at UTC timezone, thus failed with

> Path does not exist: file:/home/runner/work/kyuubi/kyuubi/kyuubi-server/target/tmp/kyuubi-9be91795-ccaa-4905-b249-8d147597f34f/kyuubi_operation/day=20240825

let me re-trigger it.

**Comment by pan3793** [comment]
Merged to master

**Comment by pan3793** [comment]
Given this causes significant perf regression(see https://github.com/apache/kyuubi/issues/6796) and the newly added functionality is not completed, let me revert it first.

**Pull request closed without merging**